NAudio/Wave/WaveOutputs/DirectSoundOut.cs

```diff
@@ -550,6 +550,41 @@ private void StopPlayback()
             {
                 if (secondaryBuffer != null)

```

Comment:
might be nice to factor this whole if statement into a helper method - e.g. CleanUpSecondaryBuffer